tests/jobs/scala/src/main/scala/KafkaCassandraStream.scala

```diff
@@ -0,0 +1,385 @@
+        .appName("Kafka Cassandra Stream")
+        .getOrCreate()
+
+      val streamingContext = new StreamingContext(spark.sparkContext, Seconds(2))
+      val cassandra = CassandraConnector(spark.sparkContext.getConf)
+      val keyspaceName = "kafka_cassandra_stream"
+      val tableName = "records"
+
+      cassandra.withSessionDo { session =>
+        session.execute(s"""DROP KEYSPACE IF EXISTS $keyspaceName""")
+        session.execute(
+          s"""CREATE KEYSPACE $keyspaceName WITH REPLICATION = {
+          'class': 'SimpleStrategy',
+          'replication_factor': 3
+        }""")
+        session.execute(
+          s"""
+          CREATE TABLE IF NOT EXISTS $keyspaceName.$tableName (
+            id uuid PRIMARY KEY,
+            first_name text,
+            last_name text,
+            email_address text,
+            address text,
+            city text
+          );"""
+        )
+      }
+
+      val recordsRdd = normalRDD(spark.sparkContext, numberOfRecords, numberOfPartitions).map(_ => randomEmployee(minColumnsLength, maxColumnsLength))
+      // TODO: input stream source should be Kafka.
+      val recordsStream = new ConstantInputDStream(streamingContext, recordsRdd)
+      val employeeColumns = SomeColumns("id", "first_name", "last_name", "email_address", "address", "city")
+
+      recordsStream.foreachRDD { rdd =>
+        println(s"Writing batch of size ${rdd.count} to Cassandra...")
+        val t0 = System.nanoTime()
+        rdd.saveToCassandra(keyspaceName, tableName, employeeColumns)
+        val t1 = System.nanoTime()
+        println(s"Wrote batch in ${(t1 - t0) / 1000 / 1000} ms")
+
+        // TODO: remove this when not using ConstantInputDStream anymore.
+        streamingContext.stop() // we only need to go through the ConstantInputDStream once.
+      }
+
+      streamingContext.start()
+      streamingContext.awaitTermination()
+    }
+  }
+}
+
+
+object KafkaRandomFeeder {
+  def main(args: Array[String]): Unit = {
+
+    val config : Config = Config()
+
+    if (!getParser("KafkaRandomFeeder").parse(args, config).isDefined) {
+      println("Error: Bad arguments")
+      System.exit(1)
+    }
+
+    val sparkBuilder = SparkSession.builder().appName(config.appName)
+
+    val spark = sparkBuilder
+      .master("local[2]")
+      .getOrCreate()
+
+    val streamingContext = new StreamingContext(spark.sparkContext, Seconds(config.batchSizeSeconds))
+
+    val wordsPerSecond = config.wordsPerSecond
+    val stream = streamingContext.receiverStream(new RandomWordReceiver(wordsPerSecond))
+
+
+    val kafkaProperties = getKafkaProperties(config.brokers, config.kerberized)
+    val topic : String = config.topics
+
+    stream.foreachRDD { rdd =>
+      rdd.foreachPartition { p =>
+
+        val producer = new KafkaProducer[String, String](kafkaProperties)
+        p.foreach { r =>
+          println(s"Writing ${r} to Kafka")
+          val msg = new ProducerRecord[String, String](topic, null, r)
```

Comment:
Personally, I try to stay away from single char variable names. We have to hold so much in our heads as is